contracts/PBAB+Collabs/GenArt721CoreV2_PBAB_FLEX.sol

```diff
@@ -0,0 +1,893 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../interfaces/0.8.x/IRandomizer.sol";
+import "../interfaces/0.8.x/IGenArt721CoreV2_PBAB.sol";
+
+import "@openzeppelin-4.5/contracts/utils/Strings.sol";
+
+import "@openzeppelin-4.5/contracts/token/ERC721/ERC721.sol";
+
+pragma solidity 0.8.9;
+
+/**
+ * @title Powered by Art Blocks ERC-721 core contract.
+ * @author Art Blocks Inc.
+ */
+contract GenArt721CoreV2_PBAB_FLEX is ERC721, IGenArt721CoreV2_PBAB {
+    /// randomizer contract
+    IRandomizer public randomizerContract;
+
+    struct Project {
+        string name;
+        string artist;
+        string description;
+        string website;
+        string license;
+        string projectBaseURI;
+        uint256 invocations;
+        uint256 maxInvocations;
+        string scriptJSON;
+        mapping(uint256 => string) scripts;
+        uint256 scriptCount;
+        string ipfsHash;
+        bool active;
+        bool locked;
+        bool paused;
+        bool externalAssetDependenciesLocked;
+        uint256 externalAssetDependencyCount;
+        mapping(uint256 => ExternalAssetDependency) externalAssetDependencies;
+    }
+
+    event ExternalAssetDependencyUpdated(
```

Comment:
talked about this with @yoshiwarab  and we DO want to use an interface, for the events and the functions we call in the new subgraph handlers. will add this next week, don't want to block our demo work for now.

Reply:
Yeah, if the event signatures are exactly the same between contracts we could technically reuse the same handler but I definitely feel more comfortable if the shared event types are explicit through a shared interface rather than implicit.

Reply:
thanks, I'm going to tag you in a PR for V3 to apply this same logic there 🙏